libcob/fileio.c

```diff
@@ -919,20 +919,34 @@ bdb_bt_compare (DB *db, const DBT *k1, const DBT *k2
 )
 {
 	const unsigned char *col = (unsigned char *)DBT_GET_APP_DATA (k1);
+#ifdef USE_BDB_KEYDIFF /* flag passed with CPPFLAGS */
+	COB_MODULE_PTR->collating_sequence = col;
```

Comment:
This changes the current module's collating sequence; we don't want that.

Reply:
agreed, we also don't want cob_fields to be used here - so extract the logic that only operates on data + size + col from cob_cmp_alnum and locally export it, then calling this here

Reply:
That's still open... just drop that line